packages/react-meteor-data/suspense/useTracker.ts

```diff
@@ -0,0 +1,305 @@
+import { Tracker } from 'meteor/tracker'
+import { EJSON } from 'meteor/ejson'
+import { DependencyList, useEffect, useMemo, useReducer, useRef } from 'react'
+import { Meteor } from 'meteor/meteor'
+
+function checkCursor(data: any): void {
+  let shouldWarn = false
+  if (Package.mongo && Package.mongo.Mongo && data && typeof data === 'object') {
+    if (data instanceof Package.mongo.Mongo.Cursor) {
+      shouldWarn = true
+    } else if (Object.getPrototypeOf(data) === Object.prototype) {
+      Object.keys(data).forEach((key) => {
+        if (data[key] instanceof Package.mongo.Mongo.Cursor) {
+          shouldWarn = true
+        }
+      })
+    }
+  }
+  if (shouldWarn) {
+    console.warn(
+      'Warning: your reactive function is returning a Mongo cursor. ' +
+      'This value will not be reactive. You probably want to call ' +
+      '`.fetch()` on the cursor before returning it.'
+    )
+  }
+}
+
+export const cache = new Map<string, Cache>()
+
+interface Cache {
+  deps: EJSON[]
+  promise: Promise<unknown>
+  result?: unknown
+  error?: unknown
+}
+
+// Used to create a forceUpdate from useReducer. Forces update by
+// incrementing a number whenever the dispatch method is invoked.
+const fur = (x: number): number => x + 1
+const useForceUpdate = () => useReducer(fur, 0)[1]
+
+export type IReactiveFn<T> = (c?: Tracker.Computation) => Promise<T>
+
+export type ISkipUpdate<T> = <T>(prev: T, next: T) => boolean
+
+interface TrackerRefs {
+  computation?: Tracker.Computation
+  isMounted: boolean
+  trackerData: any
+}
+
+function resolveAsync(key: string, promise: Promise<unknown>, deps: DependencyList = []) {
+  const cached = cache.get(key) ?? false
+  useEffect(() => {
+    return () => {
+      setTimeout(() => {
+        // maybe verify deps?
+        if (cached) cache.delete(key)
+      }, 0)
+    }
+  }, [cached, key, ...deps])
+  if (promise === null) return null
+  if (cached) {
+    if ('error' in cached) throw cached.error
+    if ('result' in cached) return cached.result
+    throw cached.promise
+  }
+
+  const entry: Cache = {
+    deps,
+    promise: new Promise((resolve, reject) => {
+      promise
+        .then((result: any) => {
+          entry.result = result
+          resolve(result)
+        })
+        .catch((error: any) => {
+          entry.error = error
+          reject(error)
+        })
+        .finally(() => {
+          // if 0 recursion :/
+          // also forcing an update thigs when out of hand
+          setTimeout(() => {
+            cache.delete(key)
+          }, 10)
+        })
```

Comment:
And is it needed? I don't understand what for, actually.

Reply:
After a while, we do need to remove the old cache entry and trigger a rerender, don't we? without this piece, it is not reactive

Reply:
Actually, I was wrong in this one. The setTimout + remove cache key was supposed to be in the return of the value